File: POO/CS/Bank/Bank/Business/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bank.Business
{
    class Product
    {
        private long barCode;

        public long BarCode
        {
            get { return barCode; }
            set { if (value < Math.Abs(1_000_000_000L)) throw new InvalidOperationException("You can't define a bar code with less than 10 digits."); barCode = value; }
        }

        private String description;

        public String Description
        {
            get { return description == null ? "<Empty>" : description; }
            set { if (value.Length < 5) throw new InvalidOperationException("You can't define a description with less than 5 characters."); description = value; }
        }

        private double price;

        public double Price
        {
            get { return price; }
            set { if (value <= 0) throw new InvalidOperationException("The value for price must be a positive value."); price = value; }
        }

        private String category;

        public String Category
        {
            get { return category == null ? "<Empty>" : category; }
            set { if (value.Length < 5) throw new InvalidOperationException("You can't define a category with less than 10 characters.");  category = value; }
        }

        private bool perishable;

        public bool Perishable
        {
            get { return perishable; }
            set { perishable = value; }
        }

        public Product()
        {
            // Standard constructor
        }
        public Product(long BarCode, String Description, double Price, String Category, bool Perishable) : base()
        {
            this.BarCode = BarCode;
            this.Description = Description;
            this.Price = Price;
            this.Category = Category;
            this.Perishable = Perishable;
        }
        // Overwritten methods
        public override string ToString()
        {
            return String.Format("\tProduct description:\nBarcode: {0,-15}\nDescription: {1}\nPrice: {2:C2}\nPerishable: {3}", BarCode, Description, Price, Category, Perishable ? "Yes" : "No");
        }
    }
}

File: POO/J/Bank/src/br/senai/sp/informatica/oop/bank/Product.java
package br.senai.sp.informatica.oop.bank;
import java.text.NumberFormat;
public class Product {
	private static NumberFormat formatter = NumberFormat.getCurrencyInstance();
	private long barCode;
	private String description;
	private double price;
	private String category;
	private boolean perishable;
	public Product() {
		// Standard constructor
	}
	public Product(long barCode, String description, double price, String category, boolean perishable) {
		super();
		setBarCode(barCode);
		setDescription(description);
		setPrice(price);
		setCategory(category);
		setPerishable(perishable);
	}
	public long getBarCode() {
		return barCode;
	}
	public void setBarCode(long barCode) {
		if(barCode < Math.abs(1_000_000_000))
			throw new IllegalArgumentException("You can't define a bar code with less than 10 digits.");
		this.barCode = barCode;
	}
	public String getDescription() {
		return description == null ? "<Empty>" : description;
	}
	public void setDescription(String description) {
		if(description.length() < 5)
			throw new IllegalArgumentException("You can't define a description with less than 5 characters.");
		this.description = description;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		if(price <= 0)
			throw new IllegalArgumentException("The value for price must be a positive value.");
		this.price = price;
	}
	public String getCategory() {
		return category == null ? "<Empty>" : category;
	}
	public void setCategory(String category) {
		if(category.length() < 5)
			throw new IllegalArgumentException("You can't define a category with less than 10 characters.");
		this.category = category;
	}
	public boolean isPerishable() {
		return perishable;
	}
	public void setPerishable(boolean perishable) {
		this.perishable = perishable;
	}
	// Overwritten methods
	@Override
	public String toString() {
		return String.format("\tProduct description:\nBarcode: %-15d\nDescription: %s\nPrice: %s\nCategory: %s\nPerishable: %s", getBarCode(), getDescription(), formatter.format(getPrice()), getCategory(), isPerishable() ? "Yes" : "No");
	}
}
